pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -17,125 +17,112 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 
 /**
  * Various utilities to be used in implementing equals and hashCode.
  *
+ * Keep isEqual() for primitive type for historical reason, but try not using them.
```

Comment:
It's not if you consider floating point inequalities (see the other comment). For example, this would be the correct implementation of equals:
```
// Check for instance equality, nulls, etc...
return isEqual(anInt, other.anInt) && isEqual(aFloat, other.aFloat);
```
with the expanded version being less readable:
```
return anInt == other.anInt && Float.floatToIntBits(aFloat) == Float.floatToIntBits(other.aFloat);
```
or even subtly incorrect:
```
return anInt == other.anInt && aFloat == other.aFloat;
```